cli/src/init.rs

```diff
@@ -624,9 +635,46 @@ pub fn generate_grammar_files(
                 Ok(())
             })?;
 
-            missing_path(repo_path.join("setup.py"), |path| {
-                generate_file(path, SETUP_PY_TEMPLATE, language_name, &generate_opts)
-            })?;
+            missing_path_else(
+                repo_path.join("setup.py"),
+                allow_update,
+                |path| generate_file(path, SETUP_PY_TEMPLATE, language_name, &generate_opts),
+                |path| {
+                    let mut contents = fs::read_to_string(path)?;
+                    if contents.contains("sources.extend") || !contents.contains("egg_info") {
+                        contents = contents
+                            .replace("sources.extend", "sources.append")
+                            .replace(
+                                "from setuptools.command.build import build\n",
+                                indoc! {"
+                                from setuptools.command.build import build
+                                from setuptools.command.egg_info import egg_info
+                                "},
+                            )
+                            .replace(
+                                "setup(\n",
+                                indoc! {r#"
+                                class EggInfo(egg_info):
+                                    def find_sources(self):
+                                        super().find_sources()
+                                        self.filelist.recursive_include("queries", "*.scm")
+                                        self.filelist.include("src/tree_sitter/*.h")
+
+
+                                setup(
+                                "#},
+                            )
+                            .replace(
+                                "\"bdist_wheel\": BdistWheel\n",
+                                // XXX: indoc drops the indent for some reason
```

Comment:
Yeah, because indoc trims the first line's leading space and uses that as a guide for indenting, so the second line isn't indented as the first line wasn't. The comment isn't needed